File: 3532.path-existence-queries-in-a-graph-i/solution.kt
class Solution {
    fun pathExistenceQueries(n: Int, nums: IntArray, maxDiff: Int, queries: Array<IntArray>): BooleanArray {
        val group = IntArray(n)
        var currentGroup = 0

        for (i in 1 until n) {
            if (nums[i] - nums[i - 1] > maxDiff) {
                currentGroup++
            }
            group[i] = currentGroup
        }

        return queries.map { (u, v) -> group[u] == group[v] }.toBooleanArray()
    }
}

File: 3532.path-existence-queries-in-a-graph-i/solution.go
func pathExistenceQueries(n int, nums []int, maxDiff int, queries [][]int) []bool {
    group := make([]int, n)
    g := 0

    for i := 1; i < n; i++ {
        if nums[i] - nums[i-1] > maxDiff {
            g++
        }
        group[i] = g
    }

    res := make([]bool, len(queries))
    
    for i, q := range queries {
        res[i] = group[q[0]] == group[q[1]]
    }
    return res
}